File: php/Fermfactor.php
<?php
/**
 * Created by PhpStorm.
 * User: pawan
 * Date: 11/7/17
 * Time: 12:02 PM
 * https://codefights.com/challenge/admES7uudjBkBNSB3
 *
 */

function fermactor($n) {
    $a = $n**.5|1;
    $b = 1.1;
    while((int)$b!=$b)
        $b = ($a++**2-$n)**.5;
    return [$a-1,$b];
}

$n=15;
print_r(fermactor($n));

File: python/FermFactor.py
# https://codefights.com/challenge/admES7uudjBkBNSB3
# https://en.wikipedia.org/wiki/Fermat%27s_factorization_method
# http://www.ams.org/journals/mcom/1974-28-126/S0025-5718-1974-0340163-2/S0025-5718-1974-0340163-2.pdf

def fermactor(n):
    a = int(n ** .5)
    b = .1
    while (int(b) != b):
        a += 1
        b = (a ** 2 - n) ** .5
    return [a, b]


n = 15
print(fermactor(n))
